src/rust/engine/src/context.rs

```diff
@@ -389,14 +389,26 @@ impl Core {
     )
     .map_err(|e| format!("Could not initialize Store: {:?}", e))?;
 
+    let store = if (exec_strategy_opts.remote_cache_read || exec_strategy_opts.remote_cache_write)
```

Comment:
Due to the lack of distinct types on `Store` (which I think that we agree is necessary for now), this is challenging to review.

One way to help clear this up would be to ensure that we're _always_ passing two values around: the `full_store` and the `local_only_store` (names TBD, but consistency is good). That would apply everywhere, but on this line, it would concretely mean having:
```
let (full_store, local_only_store) = ...
```
...which might be identical in some cases, or different in some cases.

Does that make sense?

Reply:
Passing around two same-typed values seems even more unwieldly to me.

Doing that forces every component receiving two values to have to choose which one makes sense for it. This change seemed more straight-forward to me, the local store is the only Store exposed throughout the code base (other than in remote cache code) when in eager caching mode, otherwise it is the same Store as always which preserves existing semantics.

Combined with the integration test (which necessarily triggers the intrinsic to download pex) demonstrates the fix works.